sys-fs/zfs-kmod/zfs-kmod-9999.ebuild

```diff
@@ -200,4 +200,27 @@ pkg_postinst() {
 		ewarn
 		ewarn "Refer to /usr/share/zfs/compatibility.d/grub2 for list of features."
 	fi
+
+	# check for old module layout
+	# new files are just extra/{spl,zfs}.ko
+	local ko kopath oldmodules=(
+		avl/zavl.ko
+		icp/icp.ko
+		lua/zlua.ko
+		nvpair/znvpair.ko
+		spl/spl.ko
+		unicode/zunicode.ko
+		zcommon/zcommon.ko
+		zfs/zfs.ko
+		zstd/zzstd.ko
+	)
+
+	for ko in "${oldmodules[@]}"; do
+		kopath="${EROOT}/lib/modules/${KV_FULL}/extra/${ko}"
+		if [[ -f "${kopath}" ]]; then
+			ewarn "Found previously installed zfs module ${ko} for current kernel ${KV_FULL}"
+			rm -v "${kopath}" || die
```

Comment:
Luckily, the list of suffixes is so small that we can just list them, I think (gz, lz4, xz)? Would have to check.